src/Illuminate/Console/Application.php

```diff
@@ -79,33 +80,41 @@ public function __construct(Container $laravel, Dispatcher $events, $version)
 
         $this->events->dispatch(new ArtisanStarting($this));
 
+        $this->rerouteSymfonyEvents();
+
         $this->bootstrap();
     }
 
     /**
-     * {@inheritdoc}
+     * Reroute specific Symfony events to the Laravel counterpart.
      *
-     * @return int
+     * @return void
      */
-    public function run(InputInterface $input = null, OutputInterface $output = null): int
+    protected function rerouteSymfonyEvents()
     {
-        $commandName = $this->getCommandName(
-            $input = $input ?: new ArgvInput
-        );
-
-        $this->events->dispatch(
-            new CommandStarting(
-                $commandName, $input, $output = $output ?: new BufferedConsoleOutput
-            )
-        );
-
-        $exitCode = parent::run($input, $output);
-
-        $this->events->dispatch(
-            new CommandFinished($commandName, $input, $output, $exitCode)
-        );
+        $dispatcher = new EventDispatcher();
+        $this->setDispatcher($dispatcher);
+
+        $dispatcher->addListener(ConsoleEvents::COMMAND, function (ConsoleCommandEvent $event) {
```

Comment:
optionally this can be changed to `$dispatcher->addSubscriber(...)` and move this logic to a separate class